src/Lykke.SettingsReader/SettingsProcessor/SettingsProcessor.cs

```diff
@@ -111,5 +118,146 @@ private static string ConcatPath(string path, string propertyName)
         {
             return $"{path}.{propertyName}".Trim('.');
         }
+
+        private static void ProcessChecks<T>(T model)
+        {
+            var result = new StringBuilder();
+            var attributeTypes = new List<Type>
+            {
+                typeof(HttpCheckAttribute),
+                typeof(TcpCheckAttribute),
+                typeof(AmqpCheckAttribute)
+            };
+
+            var properties = (from p in model.GetType().GetTypeInfo().GetProperties()
+                where p.CanWrite && p.CanRead && attributeTypes.Any(item => Attribute.IsDefined(p, item)) select p).ToList();
+
+            if (!properties.Any())
+                return;
+
+            Console.WriteLine("Checking services");
+
+            foreach (var property in properties)
+            {
+                string value = property.GetValue(model).ToString();
+                string url = string.Empty;
+                string address;
+                int port;
+                bool checkResult = false;
+
+                var checkAttribute = property.GetCustomAttribute(typeof(Attribute));
+
+                switch (checkAttribute)
```

Comment:
Since these checks are extension point, will be better to introduce ```ISettingsFieldValidator``` or ```ISettingsFieldChecker``` or something like this. This interface could contain single method like ```string Check/Validate(object model, object value)``` which could return validation result string. And add implementation of this interface for each check attribute. To exclude any pattenr matching switch on the attribute type. Will be better to add some base class for the check attribute with abstract method ```ISettingsFieldChecker CreateChecker()```. In this manner, new check can be added easy and without existing code modifications.